application/Dockerfile

```
@@ -9,6 +9,7 @@ RUN python -m pip install --upgrade --no-cache-dir \
     pip>=20.1.1 \
     setuptools>=49.1.0 \
     wheel>=0.34.2
+RUN python -m pip wheel -w /wheels -r requirements.txt
 RUN python -m pip wheel -w /wheels .
```

```suggestion
RUN python -m pip wheel -w /wheels -r requirements.txt .
```

---

This is what I had originally but it continues the ```-r``` and tries installing the dir as a requirement.

---

I’m getting an error for `RUN python -m pip wheel -w /wheels -r requirements.txt .` when I try this locally:

```text
ERROR: Can’t verify hashes for these file:// requirements because they point to directories:
    file:///usr/src/burnham
```

Looks like we cannot combine a requirements file that contains hashes and a local directory.

I suggest we use the following two steps:

```text
RUN python -m pip wheel -w /wheels -r requirements.txt
RUN python -m pip wheel -w /wheels —no-deps .
```

This way we install the packages in the requirements file with hashes and build wheels for them and then build a wheel only for burnham in the next step.

---

Oh wait, that wasn't my recommendation! It was to put `.` **inside** the requirements.txt and not have separate calls to `pip wheel`. pip does the right thing and then first resolves the dependency chain of the `burnham` package in addition to those requirements listed in the requirements file.

The call would just be 

```suggestion
RUN python -m pip wheel --wheel-dir /wheels -r requirements.txt
```

---

Hm, unless that doesn't work for requirements files with hashes as @hackebrot said in https://github.com/mozilla/burnham/pull/78#discussion_r500800987. So calling it twice is probably the easiest to do this, sadly.